tests/test_messaging.py

```diff
@@ -587,6 +587,28 @@ def test_invalid_thread_id(self, data):
         expected = 'Aps.thread_id must be a string.'
         assert str(excinfo.value) == expected
 
+    @pytest.mark.parametrize('data', ['', list(), tuple(), True, False, 1, 0, ])
+    def test_invalid_custom_data_dict(self, data):
+        if isinstance(data, dict):
+            return
+        aps = messaging.Aps(custom_data=data)
+        excinfo = self._check_aps(aps)
```

Comment:
I have a slight preference, here and elsewhere to inline this check, and get the with .. raises in the same test function as it is easier to read, and only 2 lines longer. But this is ok too. (maybe call it _check_aps_exception())

Reply:
Refactored as suggested.